### PR TITLE
Opportunity of injecting services into Resource.

### PR DESCRIPTION
**Pull request by magnitronus**
```ts
export class UnitRes extends Resource {
    constructor(
            http: Http,
            injector: Injector,
            private _myService: MyService
        ) {
            super(http, injector);
        }
 }
```
works now

### CODE REVIEW

File: lib/ResourceProviders.ts [hunk 1]
```diff
@@ -17,11 +18,17 @@ export class ResourceProviders {
       this.providers[subSet] = [];
     }
 
+    let deps: any[] = Reflect.getMetadata("design:paramtypes", resource)
```
Comment (troyanskiy) [inline lib/ResourceProviders.ts]: semicolon is missing

File: lib/ResourceProviders.ts [hunk 2]
```diff
@@ -17,11 +18,17 @@ export class ResourceProviders {
...
+
+    if (deps.length==0) {
```
Comment (troyanskiy) [inline lib/ResourceProviders.ts]: Please use === instead of ==

File: lib/ResourceProviders.ts [hunk 3]
```diff
@@ -17,11 +18,17 @@ export class ResourceProviders {
...
+      deps = [Http, Injector]
```
Comment (troyanskiy) [inline lib/ResourceProviders.ts]: semicolon is missing

File: lib/ResourceModel.ts [hunk 4]
```diff
@@ -5,16 +5,17 @@ import {Resource} from './Resource';
...
+    return function (target: Type<ResourceModel>) {
+      let providers:any[] = [];
+      if (params) {
+        providers = params.providers || [];
+      }
 
-    (<any>Reflect).defineMetadata('providers', providers, target);
-  };
+      (<any>Reflect).defineMetadata('providers', providers, target);
+    };
```
Comment (troyanskiy) [inline lib/ResourceModel.ts]: I think it's extra tab added for that `return function`

File: lib/ResourceProviders.ts (hunk 1 above)
Comment (magnitronus) [inline lib/ResourceProviders.ts]: Ok. Single quotes in google style.
It seems for me, is not necessary to use <any> for Reflect. I just added `import 'reflect-metadata';` for correct  type definition. What do you think about it?

File: lib/ResourceProviders.ts (hunk 2 above)
Comment (magnitronus) [inline lib/ResourceProviders.ts]: fixed.

File: lib/ResourceProviders.ts (hunk 3 above)
Comment (magnitronus) [inline lib/ResourceProviders.ts]: fixed

File: lib/ResourceModel.ts (hunk 4 above)
Comment (magnitronus) [inline lib/ResourceModel.ts]: fixed

File: lib/ResourceProviders.ts (hunk 1 above)
Comment (troyanskiy) [inline lib/ResourceProviders.ts]: Yes, it's not necessary, It's just WebStrom complaining... I have added `import 'reflect-metadata';` into `Resource.ts` and it was enough for ts compiler.

**Comment by troyanskiy** [comment]
LGTM
Thanks a lot!

**Pull request merged**